scripts/ci/pre_commit/pre_commit_check_pre_commit_hook_names.py

```diff
@@ -50,16 +57,112 @@ def main() -> int:
     return retval
 
 
-def get_errors(content, max_length):
+def get_errors_and_ids(content, max_length):
     errors = []
+    ids = set()
     for repo in content['repos']:
         for hook in repo['hooks']:
+            if 'id' in hook:
+                id = hook['id']
+                ids.add(id)
             if 'name' not in hook:
                 continue
             name = hook['name']
             if len(name) > max_length:
                 errors.append(name)
-    return errors
+    return errors, list(ids)
+
+
+def render_template(
+    searchpath: Path,
+    template_name: str,
+    context: Dict[str, Any],
+    extension: str,
+    autoescape: bool = True,
+    keep_trailing_newline: bool = False,
+) -> str:
+    """
+    Renders template based on its name. Reads the template from <name>_TEMPLATE.md.jinja2 in current dir.
+    :param template_name: name of the template to use
+    :param context: Jinja2 context
+    :param extension: Target file extension
+    :param autoescape: Whether to autoescape HTML
+    :param keep_trailing_newline: Whether to keep the newline in rendered output
+    :return: rendered template
+    """
+    import jinja2
+
+    template_loader = jinja2.FileSystemLoader(searchpath=searchpath)
+    template_env = jinja2.Environment(
+        loader=template_loader,
+        undefined=jinja2.StrictUndefined,
+        autoescape=autoescape,
+        keep_trailing_newline=keep_trailing_newline,
+    )
+    template = template_env.get_template(f"{template_name}_TEMPLATE{extension}.jinja2")
+    content: str = template.render(context)
+    return content
+
+
+def get_precommit_jinja_context(pre_commit_ids: List) -> Dict[str, List]:
```

Comment:
This could be in-lined (there is no need to separate it to a function. Simple:

```
template.render(context = { "PRE_COMMIT"; pre_commit_ids }
```

Should be enough (and instead of passing context you should bass pre_commit_ids - this is also more "logica" approach - context is "jinja" term where "pre_commit_ids" is logical data structure).